IDEAS/Examples/IBPSA/SingleZoneResidentialHydronic.mo

```diff
@@ -75,14 +76,16 @@ model SingleZoneResidentialHydronic
     "Constant supply temperature set point"
     annotation (Placement(transformation(extent={{-40,70},{-20,90}})));
   Utilities.IO.SignalExchange.Read outputQ(description="Thermal power", KPIs=
-        IDEAS.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.GasPower)
+        IDEAS.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.GasPower,
+    y(unit="W"))
     "Block for outputting the thermal power"
     annotation (Placement(transformation(extent={{60,50},{80,70}})));
   Utilities.IO.SignalExchange.Read outputP(description="Pump electrical power",
-      KPIs=IDEAS.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.ElectricPower)
+      KPIs=IDEAS.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.ElectricPower,
+    y(unit="W"))
     "Block for reading the pump electrical power"
     annotation (Placement(transformation(extent={{60,20},{80,40}})));
-  Fluid.Actuators.Valves.TwoWayTRV val(
+  TwoWayOverwriteTRV               val(
```

Comment:
see mail whether we really want this

Reply:
I see the point but I think that the possibility should be there to overwrite the actuator signal. The users can then decide whether they assume that there is a thermostatic valve (no overwrite) or that there is a motorized valve (if there is overwrite) to control the radiator.

Reply:
The point of the BOPTEST is that the _developers_ decide what the allowed control actions are. Otherwise objective comparison between approaches is not possible. Since this is a simple residential house, I'm not inclined to put a motorized valve in it. @dhblum what do you think?

Reply:
Indeed, developers decide. But both setpoints and actuator signals should be accessible to explore different control approaches. Actually, if we limit this emulator to control only the boiler supply temperature there will not be a point of an external controller at all since the valve will close when the boiler supply temperature increases.

Reply:
That's the whole point: we should provide to the user _emulators_ that represent physical reality. It's for the same reason that I argued before whether or not we should include default RBC-implementations, which I'm not fond of either, but at least the control options are still physical. I have a house that works exactly like this: I can control the supply water temperature (and pump on/off signal, which we could add here too) and nothing else.

Reply:
@JavierArroyoBastida @dhblum I propose to not allow thermostatic valve set point overrides. Do you agree?

Reply:
I still think that both, set points and actuator signals should be accessible. Actually I think that we should allow:

1. Overwrite of valve temperature set point.
2. Overwrite of valve actuator possition.
3. Overwrite of pump on/off signal (agreeing with what said in https://github.com/open-ideas/IDEAS/pull/1111#discussion_r382524789).

Reply:
Previously, the heater supply temperature was the control variable, which I presumed could be used by a controller to maintain zone temperature in the space.  With the TRV, control of zone temperature is achieved through modifying mass flow instead of supply temperature.  Adding the TRV without making it in any way programmable (at least through zone temperature setpoint adjustments) means we can't really control the system with different strategies.  As Javi pointed out, changing the water heater temperature wouldn't mean much until the TRV is saturated or at 0.  I'm not sure what the value is in such an emulator with very little controllability.  Unless the pump is controllable, in which case, the primary means of controlling zone temperature with alternative control strategies would be turning the pump on/off.  

I think controllability can still be achieved with the original formulation that Filip indicates is representative of houses in Belgium.  In that, the heater supply water temperature can be controlled to maintain house setpoint.  Is this control strategy indeed realistic?  Or is it turning the pump on/off that maintains the zone temperature comfortable?  

If the TRV is more realistic, then it should be assumed programmable in some way.  Even if it is just the temperature setpoint and not the valve position itself.  Honeywell sells TRV's with temperature setpoint programming through wireless communications: https://heatingcontrols.honeywellhome.com/products/thermostats/thermostatic-radiator-valves/Wirelessly-Controlled-TRVs/evohome-Multi-Zone-Kit/

Reply:
@dhblum thanks for commenting on this. We added a TRV to have a baseline controller as TRVs are quite common and the scheme is thus realistic. There is the other scheme that Filip pointed out: control of whater supply temperature + on/off pump control. This is more similar to the original implementation but we were missing the baseline controller. I'm not sure which one is more common, but all the residential houses I'm familiar with in Belgium and Spain work with the latter one (supply water temperature + on/off pump, **without** TRV, they use manual valves that are set to balance heat distribution to the rooms). My suggestion is to keep either one scheme or the other, which means to pick between: 

1. Original scheme: remove TRV and control supply water temperature + on/off pump, but adding a simple and realistic RBC with something like hysteresis or a heating curve.

2. We keep the TRV but we then also allow control of either valve temperature set point, or valve openning, or both. 

I think I prefer the first setting. @Mathadon what do you think?

Reply:
In practice there is often 1) one room + thermostat + radiator without TRV and 2) multiple rooms with a radiator with TRV. The thermostat will turn off the whole system when the monitored room reaches the set point and the other rooms will be controlled using a TRV. This model has only one zone, so it would make most sense to include 1) and not 2). This is the motivation for the original implementation. I still feel that it makes sense so I agree with Javi's suggestion. @dhblum ok for you?

Reply:
Yes that sounds ok to me as well.  Thank you both.